pkg/bmh/hfs_test.go

```diff
@@ -0,0 +1,231 @@
+package bmh
+
+import (
+	"fmt"
+	"testing"
+
+	bmhv1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	defaultHFSName      = "hostfirmwaresettings-test"
+	defaultHFSNamespace = "test-ns"
+)
+
+func TestPullHFS(t *testing.T) {
+	testCases := []struct {
+		hfsName             string
+		hfsNamespace        string
+		addToRuntimeObjects bool
+		client              bool
+		expectedErrorText   string
+	}{
+		{
+			hfsName:             defaultHFSName,
+			hfsNamespace:        defaultHFSNamespace,
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedErrorText:   "",
+		},
+		{
+			hfsName:             "",
+			hfsNamespace:        defaultHFSNamespace,
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedErrorText:   "hostFirmwareSettings 'name' cannot be empty",
+		},
+		{
+			hfsName:             defaultHFSName,
+			hfsNamespace:        "",
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedErrorText:   "hostFirmwareSettings 'nsname' cannot be empty",
+		},
+		{
+			hfsName:             defaultHFSName,
+			hfsNamespace:        defaultHFSNamespace,
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedErrorText: fmt.Sprintf(
+				"hostFirmwareSettings object %s does not exist in namespace %s", defaultHFSName, defaultHFSNamespace),
+		},
+		{
+			hfsName:             defaultHFSName,
+			hfsNamespace:        defaultHFSNamespace,
+			addToRuntimeObjects: true,
+			client:              false,
+			expectedErrorText:   "hostFirmwareSettings 'apiClient' cannot be nil",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testHFS := buildDummyHFS(testCase.hfsName, testCase.hfsNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testHFS)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		hfsBuilder, err := PullHFS(testSettings, testCase.hfsName, testCase.hfsNamespace)
+
+		if testCase.expectedErrorText == "" {
+			assert.Nil(t, err)
+			assert.Equal(t, testHFS.Name, hfsBuilder.Definition.Name)
+			assert.Equal(t, testHFS.Namespace, hfsBuilder.Definition.Namespace)
+		} else {
+			assert.EqualError(t, err, testCase.expectedErrorText)
+		}
+	}
+}
+
+func TestHFSGet(t *testing.T) {
```

Comment:
I think 1 scenario is missing for all test cases: 
buildValidHFSBuilder(clients.GetTestClients(clients.TestClientParams{})),

Reply:
@kononovn I use the `buildTestClientWithHFSScheme` function instead since if I don't add the scheme to the client I'll get `"no kind is registered for the type v1alpha1.HostFirmwareSettings in scheme \"pkg/clients/clients.go:302\""` errors instead of the expected. So it's equivalent, just adding the scheme even with no runtime objects